public/static/js/admin.js

```diff
@@ -87,6 +88,28 @@ mciModule.controller('AdminSettingsController', ['$scope','$window', 'mciAdminRe
       return;
     }
 
+    try {
+      $scope.Settings.container_pools.pools = jsyaml.safeLoad($scope.tempContainerPools);
+    } catch(e) {
+      alert("Error parsing container pools yaml: " + e);
+      return;
+    }
+
+    if ($scope.tempContainerPools === null || $scope.tempContainerPools === undefined || $scope.tempContainerPools == "") {
+      $scope.Settings.container_pools.pools = [];
+    }
+
+    // do not save settings if duplicate container pool IDs found
+    var uniqueIds = new Set()
+    for (var i = 0; i < $scope.Settings.container_pools.pools.length; i++) {
+      var p = $scope.Settings.container_pools.pools[i]
+      if (uniqueIds.has(p.id)) {
+        alert("Error saving settings: found duplicate container pool ID: " + p.id);
```

Comment:
are alerts propogated to users?

Reply:
Yes, it creates a javascript pop-up alert dialog. YAML validation for this and the plugins field has the same behavior.